### PR TITLE
Allow to disable tablist rewrite

### LINKED ISSUE #548: Paper's Player#setPlayerProfile not working when connect with Waterfall in offline mode
I use paper's `Player#setPlayerProfile()` to change skin of a player, using code below
```java
        PlayerProfile profile = player.getPlayerProfile();
        profile.setProperty(new ProfileProperty("textures", skin_value, skin_sign));
        player.setPlayerProfile(profile);
```
This WORKS without problem when connect to paper directly (offline or online mode), and also works when connect to Waterfall in online mode. But not working in offline mode.

**Steps to reproduce:**

 - write a paper plugin that run the above code on command.
 - set both paper server and waterfall to offline mode
 - connect directly to paper, skin changed(works)
 - connect to waterfall:
      the reload animation shows(same to above), but skin unchanged.

**anything else:**

not sure if it helps: if use FastLogin plugin and force the player using the real UUID (not the offline one), skin can be changed. But offline players(or cracked players) cannot change.

### PR DESCRIPTION
**Pull request by xxDark**

### CODE REVIEW

**Comment by mikroskeem** [comment]
Why is this useful?

File: BungeeCord-Patches/0054-Allow-to-disable-tablist-rewrite.patch [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+index cbcf8a24..4ea7678e 100644
+--- a/api/src/main/java/net/md_5/bungee/api/ProxyConfig.java
++++ b/api/src/main/java/net/md_5/bungee/api/ProxyConfig.java
+@@ -239,4 +239,9 @@ public interface ProxyConfig
+      * @return Should we disable entity metadata rewriting?
+      */
+     boolean isDisableEntityMetadataRewrite();
++
++    /**
++     * @return Should we disable tab list rewriting?
```
Comment (mikroskeem) [inline BungeeCord-Patches/0054-Allow-to-disable-tablist-rewrite.patch]: Javadoc comments shouldn't be like questions, they should describe things.

E.g `Whether tablist rewriting should be disabled or not` would fit better here.

File: BungeeCord-Patches/0054-Allow-to-disable-tablist-rewrite.patch [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ }
+diff --git a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
+index 1f8a2439..55245340 100644
+--- a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
++++ b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
+@@ -150,8 +150,12 @@ public class DownstreamBridge extends PacketHandler
+     @Override
+     public void handle(PlayerListItem playerList) throws Exception
+     {
+-        con.getTabListHandler().onUpdate( TabList.rewrite( playerList ) );
```
Comment (mikroskeem) [inline BungeeCord-Patches/0054-Allow-to-disable-tablist-rewrite.patch:60]: `// Waterfall` comments are missing

**Review by mikroskeem — APPROVED** [review]

**Comment by xxDark** [comment]
> Why is this useful?

Because it is useless with online mode and ip forwarding enabled**

File: BungeeCord-Patches/0054-Allow-to-disable-tablist-rewrite.patch (hunk 1 above)
Comment (xxDark) [reply BungeeCord-Patches/0054-Allow-to-disable-tablist-rewrite.patch]: Well, I looked at the comment above.
Gonna change jdocs and add comments as soon as possible

File: BungeeCord-Patches/0054-Allow-to-disable-tablist-rewrite.patch [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+-        throw CancelSendSignal.INSTANCE; // Always throw because of profile rewriting
++        boolean skipRewrites = bungee.getConfig().isDisableEntityMetadataRewrite();
```
Comment (mikroskeem) [inline BungeeCord-Patches/0054-Allow-to-disable-tablist-rewrite.patch]: `isDisableTabListRewrite()` should be here instead

**Review by mikroskeem — APPROVED** [review]

**Comment by oddlama** [comment]
Are there any updates on the issue?
I'm currently facing the same problem as described in #548, but unfortunately this still isn't merged. Is there a problem with the PR or something else preventing this from moving forward?

**Comment by electronicboy** [comment]
This was merged a few days ago, forgot to close the issue

**Pull request closed without merging**